tests/tape/test_tape.py

```diff
@@ -1014,11 +1014,13 @@ def test_depth_expansion(self):
             [
                 qml.BasisState([1, 0], wires=[0, 1]),
                 qml.StatePrep([0, 1, 0, 0], wires=[0, 1]),
+                qml.AmplitudeEmbedding([0, 1, 0, 0], wires=[0, 1]),
                 qml.PauliZ(0),
             ],
             [
                 qml.BasisStatePreparation([1, 0], wires=[0, 1]),
                 qml.MottonenStatePreparation([0, 1, 0, 0], wires=[0, 1]),
+                qml.StatePrep([0, 1, 0, 0], wires=[0, 1]),  # still a StatePrepBase :/
```

Comment:
I don't like this :(, 
StatePrepBase sub-classes shouldn't decompose to other instances of StatePrepBase since the `expand_tape_state_prep` isn't called recursively, we need to insure that these will always execute.

Reply:
I don't like this either but I don't want to change the decomposition because that breaks other things. it works correctly on the new default qubit : ) which will be default asap